### PR TITLE
Migrated to javalin 6.3.0 and openapi 6.3.0

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/GetMessageGroupsServlet.kt [hunk 1]
```diff
@@ -39,6 +38,7 @@ import io.javalin.openapi.OpenApiContent
 import io.javalin.openapi.OpenApiParam
 import io.javalin.openapi.OpenApiResponse
 import io.github.oshai.kotlinlogging.KotlinLogging
+import io.javalin.http.queryParamsAsClass
```
Comment (OptimumCode) [inline app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/GetMessageGroupsServlet.kt]: chore(non-blocking): please sort imports

File: app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/GetMessagesServlet.kt [hunk 2]
```diff
@@ -40,6 +39,7 @@ import io.javalin.openapi.OpenApiContent
 import io.javalin.openapi.OpenApiParam
 import io.javalin.openapi.OpenApiResponse
 import io.github.oshai.kotlinlogging.KotlinLogging
+import io.javalin.http.queryParamsAsClass
```
Comment (OptimumCode) [inline app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/GetMessagesServlet.kt]: chore(non-blocking): please sort imports

File: app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/HttpServer.kt [hunk 3]
```diff
@@ -160,37 +156,44 @@ class HttpServer(private val context: Context) {
...
+                micrometer.registry =
+                    PrometheusMeterRegistry(PrometheusConfig.DEFAULT, PrometheusRegistry.defaultRegistry, Clock.SYSTEM)
+                micrometer.tags = listOf(Tag.of("application", context.applicationName))
+            }
+
+            config.registerPlugin(MicrometerPlugin { micrometer ->
                 micrometer.registry =
                     PrometheusMeterRegistry(PrometheusConfig.DEFAULT, PrometheusRegistry.defaultRegistry, Clock.SYSTEM)
                 micrometer.tags = listOf(Tag.of("application", context.applicationName))
             })
```
Comment (OptimumCode) [inline app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/HttpServer.kt]: issue: you have duplicated parts. it does not look right

File: app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/HttpServer.kt [hunk 4]
```diff
@@ -281,15 +282,15 @@ class HttpServer(private val context: Context) {
...
                 convertToInstant(value)
             }
-            JavalinValidation.register(ProviderEventId::class.java, ::ProviderEventId)
-            JavalinValidation.register(SearchDirection::class.java, SearchDirection::valueOf)
-            JavalinValidation.register(BookId::class.java, ::BookId)
-            JavalinValidation.addValidationExceptionMapper(javalin)
+            config.validation.register(ProviderEventId::class.java, ::ProviderEventId)
+            config.validation.register(SearchDirection::class.java, SearchDirection::valueOf)
+            config.validation.register(BookId::class.java, ::BookId)
+// FIXME:           config.validation.addValidationExceptionMapper(config)
```
Comment (OptimumCode) [inline app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/HttpServer.kt]: question: what is wrong with that line?

**Comment by Nikita-Smirnov-Exactpro** [comment]
After migration, `nullable` filed has been added for some objects' fields
[openapi-default-old.json](https://github.com/user-attachments/files/18078350/openapi-default-old.json)
[openapi-default-new.json](https://github.com/user-attachments/files/18078356/openapi-default-new.json)

**Comment by OptimumCode** [comment]
> After migration, `nullable` filed has been added for some objects' fields [openapi-default-old.json](https://github.com/user-attachments/files/18078350/openapi-default-old.json) [openapi-default-new.json](https://github.com/user-attachments/files/18078356/openapi-default-new.json)

Then we need to modify annotation to indicate whether they actually can be nullable or not

**Comment by Nikita-Smirnov-Exactpro** [comment]
> > After migration, `nullable` filed has been added for some objects' fields [openapi-default-old.json](https://github.com/user-attachments/files/18078350/openapi-default-old.json) [openapi-default-new.json](https://github.com/user-attachments/files/18078356/openapi-default-new.json)
> 
> Then we need to modify annotation to indicate whether they actually can be nullable or not

Hm ... @get:OpenApiPropertyType(definedBy = ..., nullability = Nullability.NOT_NULL) doesn't affect to the "nullable":true in the openapi-default.json

**Comment by Nikita-Smirnov-Exactpro** [comment]
> > > After migration, `nullable` filed has been added for some objects' fields [openapi-default-old.json](https://github.com/user-attachments/files/18078350/openapi-default-old.json) [openapi-default-new.json](https://github.com/user-attachments/files/18078356/openapi-default-new.json)
> > 
> > 
> > Then we need to modify annotation to indicate whether they actually can be nullable or not
> 
> Hm ... @get:OpenApiPropertyType(definedBy = ..., nullability = Nullability.NOT_NULL) doesn't affect to the "nullable":true in the openapi-default.json

Added annotations to source code.

I have applied strange approach to archive not required and not nullable
```
@get:OpenApiNullable(nullable = false) // this annotation is added to mark this field as not nullable if exist
@get:OpenApiPropertyType(definedBy = ..., nullability = Nullability.NULLABLE) // thia annotation is added to mark this field as optional 
```

[openapi-default.json](https://github.com/user-attachments/files/18109343/openapi-default.json)

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**